### PR TITLE
fix b2c unit tests

### PR DESCRIPTION
**Pull request by jennyf19**
We now don't do instance discovery for b2c fix login.microsoftonline.com test

Bogdan - I left the static list because the AadAuthority list is also static, and the fix I did here fixed the test...I can still remove the static, but was originally trying to follow what we do in the other class. Let's talk tomorrow.

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Unit.net45/Resources/B2CNoTenantIdTestData.txt [hunk 1]
```diff
@@ -34,47 +34,47 @@
...
-    "environment": "login.windows.net",
+    "environment": "login.microsoftonline.com",
     "realm": "iosmsalb2c.onmicrosoft.com",
     "expires_on": "1538805103",
     "cached_at": "1538801503",
     "client_id": "0a7f52dd-260e-432f-94de-b47828c3f372",
     "home_account_id": "ad020f8e-b1ba-44b2-bd69-c22be86737f5-b2c_1_signin.ba6c0d94-a8da-45b2-83ae-33871f9c2dd8"
   },
-  "id_token_cache_key": "ad020f8e-b1ba-44b2-bd69-c22be86737f5-b2c_1_signin.ba6c0d94-a8da-45b2-83ae-33871f9c2dd8-login.windows.net-idtoken-0a7f52dd-260e-432f-94de-b47828c3f372-iosmsalb2c.onmicrosoft.com-",
+  "id_token_cache_key": "ad020f8e-b1ba-44b2-bd69-c22be86737f5-b2c_1_signin.ba6c0d94-a8da-45b2-83ae-33871f9c2dd8-login.microsoftonline.com-idtoken-0a7f52dd-260e-432f-94de-b47828c3f372-iosmsalb2c.onmicrosoft.com-",
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit.net45/Resources/B2CNoTenantIdTestData.txt]: Why is this change required ? We should always save with PreferredCacheEnv

**Comment by jennyf19** [comment]
@bgavrilMS @MarkZuber  could you re-review as this PR now has the change to match the authorityHost used in the creation of the client application with the authorityHost used in the subsequent acquire token calls. 
Addresses part of this [issue](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/911)

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 2]
```diff
@@ -386,6 +381,11 @@ public static class MsalError
         /// </summary>
         public const string NetworkNotAvailableError = "network_not_available";
 
+        /// <summary>
+        /// The B2C authority host is not the same as the one used in the config object.
```
Comment (MarkZuber) [inline src/Microsoft.Identity.Client/MsalError.cs]: suggestion:  "...same as the one used when creating the client application"  ?

File: src/Microsoft.Identity.Client/MsalErrorMessage.cs [hunk 3]
```diff
@@ -279,6 +279,7 @@ public static string CustomWebUiRedirectUriMismatch(string expectedUri, string a
         public const string MatsAndTelemetryCallbackCannotBeConfiguredSimultaneously = "MATS cannot be configured at the same time as a TelemetryCallback is provided. These are mututally exclusive.";
         public const string AkaMsmsalnet3BreakingChanges = "See https://aka.ms/msal-net-3-breaking-changes";
 
-        public const string B2CHostNotTrusted = "The B2C authority that was used is not in the trusted host list. ";
+        public const string B2CAuthorityHostMisMatch = "The B2C authority host that was used in the config object is not the same authority host used in the AcquireToken call. " +
```
Comment (MarkZuber) [inline src/Microsoft.Identity.Client/MsalErrorMessage.cs]: same here.  i don't think the "config object" means as much to customers as it does to us.

File: tests/Microsoft.Identity.Test.Unit.net45/CacheTests/TokenCacheTests.cs [hunk 4]
```diff
@@ -388,7 +388,20 @@ public void GetAppTokenFromCacheTest()
         [TestCategory("TokenCacheTests")]
         public void DoNotSaveRefreshTokenInAdalCacheForMsalB2CAuthorityTest()
         {
-            var serviceBundle = TestCommon.CreateDefaultServiceBundle();
+            var appConfig = new ApplicationConfiguration()
```
Comment (MarkZuber) [inline tests/Microsoft.Identity.Test.Unit.net45/CacheTests/TokenCacheTests.cs]: nit: can you extract this AppConfig constructor to do all of the "null the things" and have a function that just sends in the different authority uri?  looks like this is used a few times in the tests

File: tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs [hunk 5]
```diff
@@ -1090,6 +1090,35 @@ public void B2CAcquireTokenWithValidateAuthorityTrueAndRandomAuthorityTest()
             }
         }
 
+        [TestMethod]
+        [TestCategory("B2C")]
+        public void B2CAcquireTokenAuthorityHostMisMatchErrorTest()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                PublicClientApplication app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
```
Comment (MarkZuber) [inline tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs]: nit:  can you wrap the .Create / .With like you do on 1105 so it's not so wide?

**Comment by jennyf19** [comment]
[tests passing](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=322334) except integration w/known issue @bgavrilMS  is addressing
